internal/scan-manager/scanners/common/src/main/java/org/wso2/security/tools/scanmanager/scanners/common/ScannerController.java

```diff
@@ -96,12 +127,36 @@ public ResponseEntity cancelScan(@RequestBody ScannerScanRequest scanRequest) {
             String message = "No scan running to perform cancellation.";
             responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.NOT_ACCEPTABLE.value(),
                     message), HttpStatus.BAD_REQUEST);
-            log.error(message);
-            CallbackUtil.persistScanLog(scanRequest.getJobId(), message, LogType.ERROR);
         } else {
             log.info("Invoking cancel scan API.");
-            responseEntity = scanner.cancelScan(scanRequest);
+            if (scanner.validateCancelScan(scanRequest)) {
+                if (startScanThread != null) {
+                    startScanThread.interrupt();
+                } else {
+                    log.info("There is no running scan thread to cancel.");
+                }
+
+                cancelScanThread = new Thread(() -> {
+                    stopStartScanThread();
+                    scanner.cancelScan(scanRequest);
+                }, "CancelScanThread");
+                cancelScanThread.start();
+
+                responseEntity = new ResponseEntity<>(HttpStatus.ACCEPTED);
+            } else {
+                String message = "Cancel scan request validation is failed.";
+                responseEntity = new ResponseEntity<>(new ErrorMessage(HttpStatus.BAD_REQUEST.value(),
+                        message), HttpStatus.BAD_REQUEST);
+            }
         }
         return responseEntity;
     }
+
+    private boolean stopStartScanThread() {
+
+        while (startScanThread.isAlive()) {
+            // run until the start scan thread is dead.
```

Comment:
This will consume log of CPU cycles and exhaust the CPU. Please add a sleep of at least 10 second between waits (since we there is no advantage of checking status in a higher frequency).